File: V1java/inventory/src/main/java/om/inventory/service/entity/InventoryItem.java
package om.inventory.service.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.RequiredArgsConstructor;

@Data @AllArgsConstructor @RequiredArgsConstructor
@Entity @Table(name = "inventory_items")
public class InventoryItem {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String itemSkuCode;
    private Integer quantityInStock;
}

File: V2kotlin/inventory/src/main/kotlin/om/inventory/entity/InventoryItem.kt
package om.inventory.entity

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import om.inventory.dto.InventoryItemReq
import om.inventory.dto.InventoryItemResp
import org.hibernate.annotations.UuidGenerator

@Entity @Table(name = "inventory_items")
data class InventoryItem(val skuCode: String,
                         val quantityInStock: Int) {

    @Id @UuidGenerator
    @GeneratedValue(strategy = GenerationType.UUID)
    lateinit var id: String

    fun toResp() = InventoryItemResp(id, skuCode,quantityInStock)

    companion object {
        fun fromReq(itemReq: InventoryItemReq): InventoryItem =
            InventoryItem(itemReq.skuCode,itemReq.quantityToCheck)
    }
}